stdlib/REPL/src/REPL.jl

```diff
@@ -1052,20 +1055,58 @@ function setup_interface(
             oldpos = firstindex(input)
             firstline = true
             isprompt_paste = false
-            jl_prompt_len = 7 # "julia> "
+            jl_prompt_len = length(JULIA_PROMPT)
+            pkg_prompt_len = length(PKG_PROMPT)
+            shell_prompt_len = length(SHELL_PROMPT)
+            help_prompt_len = length(HELP_PROMPT)
+            curr_prompt_len = 0
+            pasting_help = false
             while oldpos <= lastindex(input) # loop until all lines have been executed
                 if JL_PROMPT_PASTE[]
-                    # Check if the next statement starts with "julia> ", in that case
-                    # skip it. But first skip whitespace
-                    while input[oldpos] in ('\n', ' ', '\t')
+                    # Check if the next statement starts with a prompt i.e. "julia> ", in that case
+                    # skip it. But first skip whitespace unless pasting in a docstring which may have
+                    # indented prompt examples that we don't want to execute
+                    while input[oldpos] in (pasting_help ? ('\n') : ('\n', ' ', '\t'))
                         oldpos = nextind(input, oldpos)
                         oldpos >= sizeof(input) && return
                     end
                     # Check if input line starts with "julia> ", remove it if we are in prompt paste mode
                     if (firstline || isprompt_paste) && startswith(SubString(input, oldpos), JULIA_PROMPT)
                         isprompt_paste = true
                         oldpos += jl_prompt_len
-                    # If we are prompt pasting and current statement does not begin with julia> , skip to next line
+                        curr_prompt_len = jl_prompt_len
+                        transition(s, julia_prompt)
+                        pasting_help = false
+                    # Check if input line starts with "pkg> ", remove it if we are in prompt paste mode and switch mode
+                    elseif (firstline || isprompt_paste) && startswith(SubString(input, oldpos), PKG_PROMPT)
+                        isprompt_paste = true
+                        oldpos += pkg_prompt_len
+                        curr_prompt_len = pkg_prompt_len
+                        Base.active_repl.interface.modes[1].keymap_dict[']'](s, o...)
+                        pasting_help = false
+                    # Check if input line starts with "(...) pkg> ", remove it if we are in prompt paste mode and switch mode
+                    elseif (firstline || isprompt_paste) && startswith(SubString(input, oldpos), Regex("\\(.+\\) $(PKG_PROMPT)"))
+                        env_pkg_prompt = match(Regex("\\(.+\\) $(PKG_PROMPT)"), SubString(input, oldpos)).match
```

Comment:
Am I understanding correctly that this changes the prompt to be the same as the pasted prompt, but does not change the environment? It seems confusing that the pkg prompt would show something other than the currently activate environment. Can we change that, so it always displays the active environment instead, regardless of what environment this is pasted from?

Reply:
No it does what you suggest. It ignores the environment in the paste

Reply:
Ok, sorry, I misunderstood what `env_pkg_prompt` does. Just ignore what I said then.

Reply:
No worries. Thanks for the review

Reply:
It can also be `pkg> ` in some circumstances, might be nice to support that too.

Reply:
That is supported. There's another branch without the regex

Reply:
Oh, why not use the same branch with an optional match on the env in the regex?

Reply:
I'm pretty bad at regex. I'd take a suggestion

Reply:
https://regex101.com/r/OcHtZm/1

```julia
julia> r = r"^(?:\(.+\) )?pkg>"
r"^(?:\(.+\) )?pkg>"

julia> match(r, "(v1) pkg> add CSV")
RegexMatch("(v1) pkg>")

julia> match(r, "pkg> add CSV")
RegexMatch("pkg>")
```
maybe?

Reply:
added. Thanks